Vr/sqlmodel 20

vreuter opened the pull request.

donaldcampbelljr commented on pipestat/argparser.py:
Why were some constants moved here from `const.py`?

donaldcampbelljr commented on pipestat/cli.py:
```diff
@@ -35,23 +35,22 @@ def main():
         status_schema_path=args.status_schema,
         flag_file_dir=args.flag_dir,
     )
```
```
 namespace=args.namespace
 status_schema_path=args.status_schema
```
Should be removed or they will throw an error during creation of psm.

donaldcampbelljr commented on pipestat/cli.py:
```diff
@@ -35,23 +35,22 @@ def main():
         value = args.value
         if psm.schema is None:
             raise SchemaNotFoundError(msg="report", cli=True)
         result_metadata = psm.schema[args.result_identifier]
```
Now returning ParsedSchema instead of schema, should be:
`result_metadata = psm.schema.results_data[args.result_identifier]`

vreuter commented:
I removed entirely what was entirely unused (by the changes), and moved definitions for things that were used in just one file to that file.Here are some thought about cost/benefit tradeoff for this sort of thing…1. Using a constant variable rather than a hardcoded value is often nice, because it boosts maintainability by reducing the extent of what must change to change a value, and because it can bind a meaningful name to an otherwise mysterious value, giving it context/meaning. This, however, comes at the cognitive and somewhat mechanical cost of INDIRECTION: I have to look somewhere else to see the actual value of something.2. The indirection cost is elevated when I need to look in a different file to find the definition.3. If a constant is USED in only one file, there’s no upside to defining it in another file, but the indirection exacerbation is still there. Perhaps one could argue that knowing const.py exists and that hypothetical future use of the constant in another file would be simplified by being able to rely on “from const.py import *” rather than needing to know where the constant was defined to import it, but that’s often unlikely and feels like a premature optimization.5. If a constant is used in multiple places, the const.py idea starts to make more sense, but really only if there’s no “natural home” for the constant. Something like an atypical delimiter, for instance, may be perfect for this. Say you want to use “/“ rather than a more typical delimiter for parsing or writing text throughout your project. But even in the case of multi-file use of the constant, if there’s a “natural home,” or a place where something like the semantic/contextual essence of the constant is most clearly materialized or otherwise relevant, then that’s where it may make most sense to define it, as it communicates the is link to future code readers.So I tried to redistribute constant definitions according to these thoughts.Le 1 mai 2023 à 19:45, Donald C ***@***.***> a écrit :﻿
@donaldcampbelljr commented on this pull request.



In pipestat/cli.py:
> @@ -35,23 +35,22 @@ def main():
         status_schema_path=args.status_schema,
         flag_file_dir=args.flag_dir,
     )
+    types_to_read_from_json = ["object"] + list(CANONICAL_TYPES.keys())
     if args.command == REPORT_CMD:
         value = args.value
         if psm.schema is None:
             raise SchemaNotFoundError(msg="report", cli=True)
         result_metadata = psm.schema[args.result_identifier]

Now returning ParsedSchema instead of schema, should be:
result_metadata = psm.schema.results_data[args.result_identifier]

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

simeoncarstens commented:
@CodiumAI-Agent /describe

CodiumAI-Agent commented:
## Title

Vr/sqlmodel 20

___
## **Type**
Enhancement, Tests


___

## **Description**
This PR introduces significant changes and enhancements to the codebase, including:
- Refactoring of test functions in `tests/test_pipestat.py` and `tests/test_init.py` to accommodate changes in the main codebase.
- Addition of new test functions to cover new features and changes.
- Introduction of new exception classes in `pipestat/exceptions.py` for better error handling.
- Update to `pipestat/__init__.py` to import the `PipestatError` class and include it in the `__classes__` list.


___

## **Changes walkthrough**
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>test_pipestat.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        tests/test_pipestat.py<br><br>

**The test file `tests/test_pipestat.py` has been <br>significantly refactored. Changes include:
- Importing <br>additional modules and functions for testing.
- Refactoring <br>of test functions to accommodate changes in the main <br>codebase.
- Addition of new test functions to cover new <br>features and changes.
- Removal of the `TestHighlighting` <br>class and its test function, and replacement with a <br>standalone test function.
- Changes in the `TestEnvVars` <br>class to test the creation of a `PipestatManager` instance <br>from environment variables.
- Changes in the <br>`TestPipestatManagerInstantiation` class to test the <br>instantiation of the `PipestatManager` class.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/pepkit/pipestat/pull/35/files#diff-50f1931db3a4fb20aab2c349c13320b1fc9c75edea89ef8608a2d8c84905d801"> +142/-70</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>test_init.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        tests/test_init.py<br><br>

**The test file `tests/test_init.py` has been significantly <br>refactored. Changes include:
- Importing additional modules <br>and functions for testing.
- Refactoring of test functions <br>to accommodate changes in the main codebase.
- Addition of <br>new test functions to cover new features and changes.
- <br>Changes in the `TestPipestatManagerInstantiation` class to <br>test the instantiation of the `PipestatManager` class.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/pepkit/pipestat/pull/35/files#diff-056c6b0b13b9539a120e8832b32e9cf5e681eb230fda35a261267d6381a0599e"> +31/-49</a></td>

</tr>                    
</table></td></tr><tr><td><strong>Error handling</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>exceptions.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        pipestat/exceptions.py<br><br>

**The `pipestat/exceptions.py` file has been updated with new <br>exception classes:
- `NoBackendSpecifiedError` for <br>designating lack of backend specification.
- <br>`UnrecognizedStatusError` for when a value to set as status <br>isn't declared in the active status schema.
- Also, some <br>minor changes in the existing exception classes.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/pepkit/pipestat/pull/35/files#diff-1c7c1bce385f2835fc045a5f3ff57b40bb67ecc45b38f0c87da47aedf6a51024"> +21/-4</a></td>

</tr>                    
</table></td></tr><tr><td><strong>Miscellaneous</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>__init__.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        pipestat/__init__.py<br><br>

**The `pipestat/__init__.py` file has been updated to import <br>the `PipestatError` class and include it in the <br>`__classes__` list.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/pepkit/pipestat/pull/35/files#diff-23496b7519ab01559922817f354bfb1ed6296083833a361403285a119c1b3316"> +2/-1</a></td>

</tr>                    
</table></td></tr></tr></tbody></table><hr>

<details> <summary><strong>✨ Usage guide:</strong></summary><hr> 

**Overview:**
The `describe` tool scans the PR code changes, and generates a description for the PR - title, type, summary, walkthrough and labels. The tool can be triggered [automatically](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) every time a new PR is opened, or can be invoked manually by commenting on a PR.

When commenting, to edit [configurations](https://github.com/Codium-ai/pr-agent/blob/main/pr_agent/settings/configuration.toml#L46) related to the describe tool (`pr_description` section), use the following template:
```
/describe --pr_description.some_config1=... --pr_description.some_config2=...
```
With a [configuration file](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#working-with-github-app), use the following template:
```
[pr_description]
some_config1=...
some_config2=...
```


<table><tr><td><details> <summary><strong> Enabling\disabling automation </strong></summary><hr>

- When you first install the app, the [default mode](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) for the describe tool is:
```
pr_commands = ["/describe --pr_description.add_original_user_description=true" 
                         "--pr_description.keep_original_user_title=true", ...]
```
meaning the `describe` tool will run automatically on every PR, will keep the original title, and will add the original user description above the generated description. 

- Markers are an alternative way to control the generated description, to give maximal control to the user. If you set:
```
pr_commands = ["/describe --pr_description.use_description_markers=true", ...]
```
the tool will replace every marker of the form `pr_agent:marker_name` in the PR description with the relevant content, where `marker_name` is one of the following:
  - `type`: the PR type.
  - `summary`: the PR summary.
  - `walkthrough`: the PR walkthrough.

Note that when markers are enabled, if the original PR description does not contain any markers, the tool will not alter the description at all.
        


</details></td></tr>

<tr><td><details> <summary><strong> Custom labels </strong></summary><hr>

The default labels of the `describe` tool are quite generic: [`Bug fix`, `Tests`, `Enhancement`, `Documentation`, `Other`].

If you specify [custom labels](https://github.com/Codium-ai/pr-agent/blob/main/docs/DESCRIBE.md#handle-custom-labels-from-the-repos-labels-page-gem) in the repo's labels page or via configuration file, you can get tailored labels for your use cases.
Examples for custom labels:
- `Main topic:performance` - pr_agent:The main topic of this PR is performance
- `New endpoint` - pr_agent:A new endpoint was added in this PR
- `SQL query` - pr_agent:A new SQL query was added in this PR
- `Dockerfile changes` - pr_agent:The PR contains changes in the Dockerfile
- ...

The list above is eclectic, and aims to give an idea of different possibilities. Define custom labels that are relevant for your repo and use cases.
Note that Labels are not mutually exclusive, so you can add multiple label categories.
Make sure to provide proper title, and a detailed and well-phrased description for each label, so the tool will know when to suggest it.        


</details></td></tr>



<tr><td><details> <summary><strong> More PR-Agent commands</strong></summary><hr> 

> To invoke the PR-Agent, add a comment using one of the following commands:  
> - **/review**: Request a review of your Pull Request.   
> - **/describe**: Update the PR title and description based on the contents of the PR.   
> - **/improve [--extended]**: Suggest code improvements. Extended mode provides a higher quality feedback.   
> - **/ask \<QUESTION\>**: Ask a question about the PR.   
> - **/update_changelog**: Update the changelog based on the PR's contents.   
> - **/add_docs** 💎: Generate docstring for new components introduced in the PR.   
> - **/generate_labels** 💎: Generate labels for the PR based on the PR's contents.   
> - **/analyze** 💎: Automatically analyzes the PR, and presents changes walkthrough for each component.   

>See the [tools guide](https://github.com/Codium-ai/pr-agent/blob/main/docs/TOOLS_GUIDE.md) for more details.
>To list the possible configuration parameters, add a **/config** comment.   
 


</details></td></tr>

</table>

See the [describe usage](https://github.com/Codium-ai/pr-agent/blob/main/docs/DESCRIBE.md) page for a comprehensive guide on using this tool.


</details>
